atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionManager.java

```diff
@@ -429,20 +429,26 @@ <T, C extends PreCommitCondition, E extends Exception> T runTaskWithConditionRea
     void registerClosingCallback(Runnable closingCallback);
 
     /**
-     * This method can be used for direct control over the lifecycle of a batch of transactions. For example, if the
-     * work done in each given transaction is interactive and cannot be expressed as a {@link TransactionTask} ahead of
-     * time, this method allows for a long lived transaction object. For any data read or written to the transaction to
-     * be valid, the transaction must be committed by calling {@link OpenTransaction#finish(TransactionTask)} to
-     * also perform additional cleanup. Note that this does not clean up the pre commit condition associated with that
-     * task. The order of transactions returned corresponds with the pre commit conditions passed in, however there are
+     * This method can be used for direct control over the lifecycle of a batch of transactions.
+     * For example, if the work done in each given transaction is interactive and cannot be expressed as a
+     * {@link TransactionTask} ahead of time, this method allows for a long lived transaction object.
+     * <p>
+     * For any data read or written to the transaction to be valid, the transaction must be committed explicitly
+     * by the caller with {@link Transaction#commit()}.
+     * <p>
+     * Note the caller must also call {@link OpenTransaction#close()} after the transaction is committed to perform
+     * additional cleanup. {@link OpenTransaction#close} does not clean up the pre commit condition associated with that
```

Comment:
`{@link OpenTransaction#close} does not clean up the pre commit condition associated with that` - this is also weird to me - we could just register it to be cleaned up as we do on `runTaskWithConditionThrowOnConflict`?

And our main internal (and only one) consumer literally does it on `finally`, so feels confusing why we don't do it here.

Reply:
Yeah let's change this, unless I've missed something about the ordering my understanding is that it's ok? `OpenTransaction#close` is only called by the atlas proxy which wants it to be closed

Reply:
Correct. I'll change this.